File: domain/src/test/scala/net/kemitix/thorp/domain/MD5HashTest.scala
package net.kemitix.thorp.domain

import org.scalatest.funspec.AnyFunSpec

class MD5HashTest extends AnyFunSpec {

  describe("recover base64 hash") {
    it("should recover base 64 #1") {
      val rootHash = MD5HashData.Root.hash
      assertResult(MD5HashData.Root.base64)(MD5Hash.hash64(rootHash))
    }
    it("should recover base 64 #2") {
      val leafHash = MD5HashData.Leaf.hash
      assertResult(MD5HashData.Leaf.base64)(MD5Hash.hash64(leafHash))
    }
  }
}

File: domain/src/test/java/net/kemitix/thorp/domain/MD5HashTest.java
package net.kemitix.thorp.domain;

import org.assertj.core.api.WithAssertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class MD5HashTest
        implements WithAssertions {

    @Test
    @DisplayName("recover base64 hash")
    public void recoverBase64Hash() {
        assertThat(MD5HashData.Root.hash.hash64())
                .isEqualTo(MD5HashData.Root.base64);
        assertThat(MD5HashData.Leaf.hash.hash64())
                .isEqualTo(MD5HashData.Leaf.base64);
    }

    @Test
    @DisplayName("hash() strips quotes")
    public void hashStripsQuotes() {
        //given
        String dQuote = "\"";
        MD5Hash md5Hash = MD5Hash.create(dQuote + MD5HashData.Root.hashString + dQuote);
        //when
        String result = md5Hash.hash();
        //then
        assertThat(result).isEqualTo(MD5HashData.Root.hashString);
    }
}
